tests/validation/tests/v3_api/resource/terraform/longliving/aws/v2apps/main.tf

```diff
@@ -0,0 +1,137 @@
+terraform {
+    required_providers {
+        rancher2 = {
+            source  = "rancher/rancher2"
+            version = "1.10.6"
+        }
+    }
+}
+
+provider "aws" {
+    region  = "us-west-2"
+    profile = "rancher-eng"
+}
+
+provider "rancher2" {
+    api_url    = var.cattle_test_url
+    token_key  = var.admin_token
+    insecure   = true
+}
+
+locals {
+    node_config = ["controlplane", "etcd", "worker", "worker", "worker"]
+}
+
+// create custom cluster in Rancher
+resource "rancher2_cluster" "custom-cluster" {
+    name = "${var.node_name}"
+    description = "Created from Terraform"
+    rke_config {
+        addon_job_timeout = 45
+        cloud_provider {
+            name = "aws"
+            aws_cloud_provider {
+            }
+        }
+        network {
+            plugin = "canal"
+        }
+        services {
+            etcd {
+                backup_config {
+                    enabled = true
+                    retention = 3
+                    interval_hours = 6
+                    s3_backup_config {
+                        bucket_name = var.minio_bucket
+                        endpoint = var.minio_endpoint
+                        access_key = var.minio_access_key
+                        secret_key = var.minio_secret_key
+                        custom_ca = var.minio_ca
+                    }
+                }
+            }
+            kube_api {
+                secrets_encryption_config {
+                    enabled = true
+                }
+            }
+        }      
+    }
+}
+
+// Create a new rancher2 Cluster Sync for custom-cluster cluster
+resource "rancher2_cluster_sync" "custom-cluster" {
+  cluster_id        = rancher2_cluster.custom-cluster.id
+  state_confirm     = 3
+}
```

Comment:
`depends_on = [aws_instance.custom_nodes]` is needed.

Reply:
This `depends_on` will be needed anywhere we are creating custom clusters